File: examples/courseman/msa/modules/service1/enrolment-service1/src/main/java/org/jda/example/coursemanmsa/enrolment/model/Student.java
package org.jda.example.coursemanmsa.enrolment.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Transient;
import org.springframework.hateoas.RepresentationModel;

import com.fasterxml.jackson.annotation.JsonInclude;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter @Setter @ToString
@Entity
@Table(name="student")
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Student extends RepresentationModel<Student> {

	@Id
	@Column(name = "id", nullable = false)
	private String id;
	@Column(name = "name", nullable = false)
	private String studentName;
	@Column(name = "gender_name")
	private String studentGender;
	@Column(name = "dob")
	private Date studentDob;
	@Column(name = "address_id", nullable = false)
	private int addressId;
	@Column(name="email")
	private String studentEmail;
	@Column(name = "studentclass_id", nullable = false)
	private int studentclassId;
	
	@Column(name="addressname")
	private String addressName;
	
	@Column(name="studentclassname")
	private String studentClassName;
}

File: examples/courseman/fereactnative/src/data_types/Student.ts
import {Gender} from './enums/Gender';

import {Address} from './Address';
import {StudentClass} from './StudentClass';
import {Enrolment} from './Enrolment';

export interface Student {
  id: string;
  name: string;
  gender: Gender;
  dob: Date;
  address?: Omit<Address, 'student'>;
  email: string;
  studentClass?: Omit<StudentClass, 'students'>;
  enrolments: Enrolment[];
}
